### PR TITLE
[CCFPCM-643] pos refactor for SQ warning

### PR DESCRIPTION
**Pull request by chelsea-EYDS**
[CCFPCM-643](https://bcdevex.atlassian.net/browse/CCFPCM-643)

Objective: 

- refactor for: SQ cognitive complexity for pos reconciliation service

### CODE REVIEW

File: apps/backend/src/transaction/payment.service.ts [hunk 1]
```diff
@@ -221,6 +210,21 @@ export class PaymentService {
...
   }
+  /**
+   * Update payments matched on round four heuristics - calling "save" as "update" will not trigger the cascade for the many to many relations
+   * @param payments
+   * @returns
+   */
+  async updatePOSPayments(payments: PaymentEntity[]): Promise<PaymentEntity[]> {
+    try {
+      return await Promise.all(
+        payments.map((itm) => this.paymentRepo.save(itm))
```
Comment (chelsea-EYDS) [inline apps/backend/src/transaction/payment.service.ts:221]: TODO - use transaction manager

File: apps/backend/src/reconciliation/ministryHeuristics.ts [hunk 2]
```diff
@@ -0,0 +1,67 @@
+import { differenceInSeconds } from 'date-fns';
```
Comment (fwkendall) [inline apps/backend/src/reconciliation/ministryHeuristics.ts:1]: this is SO much cleaner pulled out like this

**Review by fw-noel — APPROVED** [review]
Good improvements to readability :)

File: apps/backend/src/reporting/reporting.service.ts [hunk 3]
```diff
@@ -696,7 +696,7 @@ export class ReportingService {
     qb.select('payment.status');
     qb.addSelect('COUNT(payment.status)', 'count');
     qb.leftJoin('payment.payment_method', 'payment_method');
-    qb.andWhere('payment_method.classification = :classification', {
+    qb.where('payment_method.classification = :classification', {
```
Comment (fw-noel) [inline apps/backend/src/reporting/reporting.service.ts:699]: This is fine, but FYI, `andWhere` acts like a `where` if its the only clause, so it makes it like 0.05 seconds shorter to add further clauses in the future.

No need to change :p

**Review by fw-noel — APPROVED** [review]

**Pull request merged**